controller/search_blackbox_test.go

```diff
@@ -1030,6 +1030,175 @@ func (s *searchControllerTestSuite) TestSearchByJoinedData() {
+			require.NotEmpty(t, result.Data)
+			assert.Len(t, result.Data, 4)
+			toBeFound := id.MapFromSlice(id.Slice{
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+				fxt.WorkItems[7].ID,
+				fxt.WorkItems[8].ID,
+			})
+			for _, wi := range result.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", *wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+		t.Run("with one child iteration", func(t *testing.T) {
+			filter := fmt.Sprintf(`{"iteration": "%s/**"}`, fxt.Iterations[1].ID)
+			_, result := test.ShowSearchOK(t, nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+			require.NotEmpty(t, result.Data)
+			assert.Len(t, result.Data, 6)
+			toBeFound := id.MapFromSlice(id.Slice{
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+				fxt.WorkItems[7].ID,
+				fxt.WorkItems[8].ID,
+			})
+			for _, wi := range result.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", *wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+		t.Run("with two child iteration", func(t *testing.T) {
+			filter := fmt.Sprintf(`{"iteration": "%s/**"}`, fxt.Iterations[0].ID)
+			_, result := test.ShowSearchOK(t, nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+			require.NotEmpty(t, result.Data)
+			assert.Len(t, result.Data, 9)
+			toBeFound := id.MapFromSlice(id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+				fxt.WorkItems[7].ID,
+				fxt.WorkItems[8].ID,
+			})
+			for _, wi := range result.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", *wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+
+		t.Run("without child iteration - implicit", func(t *testing.T) {
+			filter := fmt.Sprintf(`{"iteration": "%s"}`, fxt.Iterations[2].ID)
+			_, result := test.ShowSearchOK(t, nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+			require.NotEmpty(t, result.Data)
+			assert.Len(t, result.Data, 4)
+			toBeFound := id.MapFromSlice(id.Slice{
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+				fxt.WorkItems[7].ID,
+				fxt.WorkItems[8].ID,
+			})
+			for _, wi := range result.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", *wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+		t.Run("without child iteration - false option", func(t *testing.T) {
+			filter := fmt.Sprintf(`{"iteration": "%s"}`, fxt.Iterations[2].ID)
```

Comment:
I think you need to provide 'child:false' option here as the test says 'false option'. Otherwise how would this test be diffferent from the one above it, right?

Reply:
Fixed in 33d6069235502f772aee2f7e19e49c0374058894

Reply:
I reverted the commit, because it's no more relying on `false` option rather we use a `/**` suffix to get children. I will update the test description.